### PR TITLE
Take a bool for SSL domain validation

### PR DESCRIPTION
**Pull request by afilini**

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -113,20 +135,26 @@ impl Client<ElectrumPlaintextStream> {
...
+    /// `validate_domain` is `true`, validate the server's certificate.
+    pub fn new_ssl<A: ToSocketAddrsDomain>(
+        socket_addr: A,
+        validate_domain: bool,
+    ) -> Result<Self, Error> {
         let mut builder =
             SslConnector::builder(SslMethod::tls()).map_err(Error::InvalidSslMethod)?;
         // TODO: support for certificate pinning
-        if domain.is_none() {
+        if !validate_domain {
```
Comment (RCasatta) [inline src/client.rs]: nit: inverting if branch to avoid the not

File: src/client.rs [hunk 2]
```diff
@@ -49,6 +49,28 @@ macro_rules! impl_batch_call {
...
+pub trait ToSocketAddrsDomain: ToSocketAddrs {
+    /// Returns the domain, if present
+    fn domain(&self) -> Option<&str> {
+        None
+    }
+}
+
+impl ToSocketAddrsDomain for &str {
+    fn domain(&self) -> Option<&str> {
+        self.splitn(2, ':').into_iter().next()
```
Comment (RCasatta) [inline src/client.rs]: nit: clippy says you can remove into_iter()

Comment (afilini) [reply src/client.rs]: fixed in 78a8a7c

File: src/client.rs (hunk 1 above)
Comment (afilini) [reply src/client.rs]: fixed in 78a8a7c

**Comment by RCasatta** [comment]
utack  68cd825

**Pull request merged**